Add comment_command_flags and comment_package_manager params

gonzaloriestra opened the pull request:
I'm adding two optional params:
- `comment_command_flags`: to append to the global install command
- `comment_package_manager`: to use for the global install command instead of the inferred one

We want them to [simplify the snapit message in the CLI](https://github.com/Shopify/cli/pull/6329):

| Before | After |
|--------|--------|
| <img width="819" height="388" alt="28-36-uegu6-4wnyj" src="https://github.com/user-attachments/assets/aacc4ad3-11a0-443e-ad6a-b41444a445e0" /> | <img width="925" height="394" alt="28-37-ol36n-easws" src="https://github.com/user-attachments/assets/759299af-911a-405e-9829-ae5a1040133d" /> |

gonzaloriestra committed:
Add shopify_registry and package_manager params

gonzaloriestra committed:
Merge branch 'main' into registry-and-package-manager

gonzaloriestra committed:
Rename new options

alex-page commented:
@gonzaloriestra I can do a release tomorrow. Just merge to `main`

gonzaloriestra commented on action.yml:
```diff
@@ -13,6 +13,10 @@ inputs:
+  comment_package_manager:
+    description: Package manager to show for global installation (yarn/npm/pnpm). If not specified, will auto-detect based on lock files.
+  comment_shopify_registry:
+    description: Shopify NPM registry URL (e.g. https://registry.npmjs.org). Will be added as --@shopify:registry= to global installation command.
```
It's actually used to update a Shopify thing: `--@shopify:registry=${commentShopifyRegistry}`

But I'm thinking that we can rename it to something more generic like `comment_command_flags` so that we can use this way:
```
  comment_command_flags: '--@shopify:registry=https://registry.npmjs.org'
```

Does it make sense?

gonzaloriestra replied:
I guess we can, but it's better to follow the same approach as with other inputs, no? It may be useful for other projects.

alex-page replied:
https://github.com/Shopify/admin-ui-foundations-2025/blob/a015cdaa7b686621c56c7bddefe224540a134a17/.github/workflows/libraries-javascript-polaris_snapit.yml#L27-L32

alex-page replied:
I just don't fully understand the need of this request. You are adding an install flag which most people won't need because their CI or local development will already be configured to correctly get from the registry.

gonzaloriestra committed:
Generalize comment_shopify_registry to comment_command_flags

gonzaloriestra commented:
@alex-page the main branch seems to be broken. I tried first with this branch and it didn't work, although it didn't show any error.

Then I switched to main (after fixing [this typo](https://github.com/Shopify/snapit/pull/45)), and I got the same result: https://github.com/Shopify/cli/actions/runs/19293916160/job/55171040211

Could you please have a look?

alex-page committed:
Merge branch 'main' into registry-and-package-manager

alex-page commented:
> @alex-page the main branch seems to be broken. I tried first with this branch and it didn't work, although it didn't show any error.
>
> Then I switched to main (after fixing https://github.com/Shopify/snapit/pull/45), and I got the same result: https://github.com/Shopify/cli/actions/runs/19293916160/job/55171040211

It seems like it is green in CI? Is there a failing run or is it just not publishing?

gonzaloriestra commented:
@alex-page it's not failing, but not publishing. It just finishes and does nothing.

gonzaloriestra committed:
Merge branch 'main' into registry-and-package-manager

gonzaloriestra committed:
Fix typo

gonzaloriestra commented:
@alex-page there was [another typo from main](https://github.com/Shopify/snapit/pull/41/changes/6446eb83e2747834148edf3b02bb6b4cff974b95), but it's working now: https://github.com/Shopify/cli/pull/6750#issuecomment-3771945626

Can we merge it?

gonzaloriestra commented:
@alex-page could we merge this PR? I don't want to keep it open forever and the CLI pointing to this branch...

If there are release instructions somewhere, I can take care it. Thanks!

craigmichaelmartin committed:
Fix empty string fallback for comment_package_manager

When comment_package_manager input is not set, core.getInput() returns
an empty string. Using nullish coalescing (??) fails to fall back to
the auto-detected packageManager because empty string is not nullish.
This causes the install command to render as 'i -g' instead of 'npm i -g'.

Switch to logical OR (||) so empty string correctly falls through to
the detected package manager.

gonzaloriestra committed:
Merge pull request #55 from Shopify/fix-empty-package-manager-fallback

Fix empty string fallback for comment_package_manager

gonzaloriestra committed:
Remove deprecated flags
